scheduler/scheduler.go

```diff
@@ -246,6 +236,86 @@ func NewMesosSchedulerDriver(config DriverConfig) (initializedDriver *MesosSched
 	return
 }
 
+func (driver *MesosSchedulerDriver) makeWithScheduler(cs Scheduler) func(func(Scheduler)) {
```

Comment:
Me too, and I wrote it. I tried to tease things apart for readability but
it's still too complex for my taste. But it *does* solve the hanging
problems that I observed at scheduler shutdown. I can clean up the deferred
lock management so that it's less heady.

On Thu, Sep 24, 2015 at 5:59 AM, Tomás Senart <notifications@github.com>
wrote:

> In scheduler/scheduler.go
> <https://github.com/mesos/mesos-go/pull/189#discussion_r40302941>:
>
> > @@ -246,6 +236,86 @@ func NewMesosSchedulerDriver(config DriverConfig) (initializedDriver *MesosSched
> >  	return
> >  }
> >
> > +func (driver *MesosSchedulerDriver) makeWithScheduler(cs Scheduler) func(func(Scheduler)) {
>
> OMG, I think I need a vacation after before reading this function. :-)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesos/mesos-go/pull/189/files#r40302941>.
>